src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -782,10 +782,7 @@ public void DeleteTombstonesBefore(string collection, long etag, Transaction tra
                 return;
             }
 
-            foreach (var result in table.SeekBackwardFrom(_tombstonesSchema.FixedSizeIndexes["CollectionEtags"], etag))
-            {
-                table.Delete(result.Id);
```

Comment:
I was just talking with Arek about this, maybe we can track number of opened iterators and throw if the count is > 0 ? What do you think about this approach?